<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Case Control Allele Frequency Estimation • CCAFE</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Case Control Allele Frequency Estimation">
<meta property="og:description" content="Functions to reconstruct case and control AFs from summary statistics. One function uses OR, NCase, NControl, and SE(log(OR)). The second function uses OR, NCase, NControl, and AF for the whole sample.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">CCAFE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/CCAFE.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/CCAFE_CCGWAS.html">CCAFE CC-GWAS Example</a>
    </li>
    <li>
      <a href="articles/CCAFE_Extra_Details.html">CCAFE Extra Details</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/wolffha/CCAFE/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="ccafe">CCAFE<a class="anchor" aria-label="anchor" href="#ccafe"></a>
</h1></div>
<p><strong>C</strong>ase <strong>C</strong>ontrol <strong>A</strong>llele <strong>F</strong>requency (AF) <strong>E</strong>stimation R Package</p>
<p>This repository contains the source code for the CaseControlAF R package which can be used to reconstruct the allele frequency (AF) for cases and controls separately given commonly available summary statistics.</p>
<p>The package contains two functions:</p>
<ol style="list-style-type: decimal">
<li>CaseControl_AF</li>
<li>CaseControl_SE</li>
</ol>
<p>See full documentation, vignettes, and examples here: (<a href="https://wolffha.github.io/CCAFE/" class="external-link uri">https://wolffha.github.io/CCAFE/</a>)</p>
<div class="section level2">
<h2 id="download-the-package">Download the package<a class="anchor" aria-label="anchor" href="#download-the-package"></a>
</h2>
<p>To install this package using BioConductor (Not yet available):</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"CCAFE"</span><span class="op">)</span></span></code></pre></div>
<p>To download this package using <em>devtools</em> in R:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"https://github.com/wolffha/CCAFE/"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="casecontrol_af">CaseControl_AF<a class="anchor" aria-label="anchor" href="#casecontrol_af"></a>
</h2>
<p>Use this function when you have the following statistics (for each variant)</p>
<ul>
<li>Number of cases</li>
<li>Number of controls</li>
<li>Odds Ratio (OR) or beta coefficient</li>
<li>
<strong>AF</strong> (allele frequency) for the total sample (cases and controls combined)</li>
</ul>
<div class="section level3">
<h3 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h3>
<p><strong>data</strong>: a dataframe with a row for each variant and columns for OR and total AF</p>
<p><strong>N_case</strong>: an integer for the number of case samples</p>
<p><strong>N_control</strong>: an integer for the number of control samples</p>
<p><strong>OR_colname</strong>: a string containing the exact column name in ‘data’ with the OR</p>
<p><strong>AF_total_colname</strong>: a string containing the exact column name in ‘data’ with the total AF</p>
<p>Returns a dataframe with two columns: AF_case and AF_control. The number of rows is equal to the number of variants.</p>
</div>
</div>
<div class="section level2">
<h2 id="casecontrol_se">CaseControl_SE<a class="anchor" aria-label="anchor" href="#casecontrol_se"></a>
</h2>
<p>Use this function when you have the following statistics (for each variant)</p>
<ul>
<li>Number of cases</li>
<li>Number of controls</li>
<li>Odds Ratio (OR) or beta coefficient</li>
<li>
<strong>SE</strong> of the log(OR) for each variant</li>
</ul>
<p><em>Code adapted from ReACt GroupFreq function available here: (<a href="https://github.com/Paschou-Lab/ReAct/blob/main/GrpPRS_src/CountConstruct.c" class="external-link uri">https://github.com/Paschou-Lab/ReAct/blob/main/GrpPRS_src/CountConstruct.c</a>)</em></p>
<div class="section level3">
<h3 id="usage-1">Usage<a class="anchor" aria-label="anchor" href="#usage-1"></a>
</h3>
<p><strong>data</strong>: a dataframe where each row is a variant and columns for the OR, SE, chromosome, and position</p>
<p><strong>N_case</strong>: an integer for the number of case samples</p>
<p><strong>N_control</strong>: an integer for the number of control samples</p>
<p><strong>OR_colname</strong>: a string containing the exact column name in <em>data</em> with the odds ratios</p>
<p><strong>SE_colname</strong>: a string containing the exact column name in <em>data</em> with the standard errors</p>
<p><strong>position_colname</strong>: a string containing the exact column name in <em>data</em> with the positions of the variants</p>
<p><strong>chromosome_colname</strong>: a string containing the exact column name in <em>data</em> with the chromosome of the variants. Note, sex chromosomes can be either characters (‘X’, ‘x’, ‘Y’, ‘y’) or numeric where X=23 and Y=24</p>
<p><strong>sex_chromosomes</strong>: boolean, TRUE if variants from sex chromosome(s) are included in the dataset</p>
<p><strong>do_correction</strong>: boolean, TRUE if data is provided to correct the estimates using proxy MAFs</p>
<p><strong>remove_sex_chromosomes</strong>: boolean, TRUE if variants on sex chromosomes should be removed. This is only necessary if <em>sex_chromosomes</em> == TRUE and the number of XX/XY individuals per case and control sample is NOT known</p>
<p>CaseControl_SE has the following optional inputs:</p>
<p>If <em>sex_chromosomes</em> == TRUE and <em>remove_sex_chromosomes</em> == FALSE, then the following inputs are required:</p>
<p><strong>N_XX_case</strong>: the number of XX chromosome case individuals</p>
<p><strong>N_XX_control</strong>: the number of XX chromosome control individuals</p>
<p><strong>N_XY_case</strong>: the number of XY chromosome case individuals</p>
<p><strong>N_XY_control</strong>: the number of XY chromosome control individuals</p>
<p>If <em>do_correction</em> == TRUE, then data must be provided that includes harmonized data with proxy MAFs</p>
<p><strong>correction_data</strong>: a dataframe with the following EXACT column names: CHR, POS, proxy_MAF, containing data for variants harmonized between the observed and proxy datasets</p>
<p>Returns the <em>data</em> dataframe with three additional columns with names: MAF_case, MAF_control and MAF_total containing the estimated minor allele frequency in the cases, controls, and total sample. The number of rows is equal to the number of variants. If proxyMAFs_colname is not NA, will include three additional columns containing the adjusted estimated MAFs (MAF_case_adj, MAF_control_adj, MAF_total_adj)</p>
<p><strong>NOTE:</strong> This method assumes we are estimating the minor allele frequency (MAF)</p>
</div>
<div class="section level3">
<h3 id="examples-and-documentation">Examples and documentation<a class="anchor" aria-label="anchor" href="#examples-and-documentation"></a>
</h3>
<p>See full documentation, vignettes, and examples here: (<a href="https://wolffha.github.io/CaseControlAF/" class="external-link uri">https://wolffha.github.io/CaseControlAF/</a>)</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/wolffha/CCAFE/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/wolffha/CCAFE/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing CCAFE</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Hayley Wolff <br><small class="roles"> Maintainer, author </small>  </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Hayley Wolff.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
