---
title: "CCAFE_correction_limits"
author: "Hayley Stoneman"
date: "2025-02-05"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Make sure I'm using latest version of CCAFE
```{r}
#devtools::install_github("https://github.com/wolffha/CCAFE", force = T, build_vignettes = T)
library(CCAFE)
library(tidyverse)
library(cowplot)
```

```{r}
setwd("C:/Users/HayBa/Documents/Graduate School/CaseControlAF/")
```


```{r, cache=TRUE}
gnomad <- read.csv("C:/Users/HayBa/Documents/Graduate School/GeneticPredisposition/CCPM_prostate/gnomad_ukb_diab_merged_chr1.csv")
gnomad <- gnomad %>% filter(!is.na(se_AFR) & !is.na(se_EUR))
gnomad <- gnomad %>% filter(alt == REF | alt == ALT)
# Keep only SNPs that have at least one match between alt/ref alleles - down to 14,212,618

# flip alleles so alleles in gnomAD match the reported ones in PanUK Biobank
gnomad <- gnomad %>% rowwise() %>%
  mutate(flipped = ifelse(alt == REF, 1, 0)) %>%
  mutate(match_ALT = ifelse(flipped == 1, REF, ALT),
         match_AFnfe = ifelse(flipped == 1, 1-AFnfe, AFnfe),
         match_AFafr = ifelse(flipped == 1, 1-AFafr, AFafr))

# get the maf for diabetes cases in pan-uk biobank and gnomAD NFE and AFR/AFRAM
gnomad <- gnomad %>% rowwise() %>%
  mutate(maf_cases_EUR = ifelse(af_cases_EUR > 0.5, 1 - af_cases_EUR, af_cases_EUR),
         maf_cases_AFR = ifelse(af_cases_AFR > 0.5, 1 - af_cases_AFR, af_cases_AFR),
         maf_gnomad_EUR = ifelse(match_AFnfe > 0.5, 1 - match_AFnfe, match_AFnfe),
         maf_gnomad_AFR = ifelse(match_AFafr > 0.5, 1 - match_AFafr, match_AFafr),
         maf_controls_EUR = ifelse(af_controls_EUR > 0.5, 1 - af_controls_EUR, af_controls_EUR),
         maf_controls_AFR = ifelse(af_controls_AFR > 0.5, 1 - af_controls_AFR, af_controls_AFR))
```

gnomAD chr1 dataset has 1,212,618 variants
```{r}
colnames(gnomad)
```
First look at MAF bins right now
```{r}
nCase_diab = 16550
nControl_diab = 403923

gnomad$maf_total_EUR <- (gnomad$maf_cases_EUR*nCase_diab + gnomad$maf_controls_EUR*nControl_diab)/
  (nCase_diab + nControl_diab)
gnomad$maf_total_AFR <- (gnomad$maf_cases_AFR*668 + gnomad$maf_controls_AFR*5956)/
  (668+5956)

breaks <- seq(0, 0.5, by = 0.1)
gnomad$bin_EUR <- cut(gnomad$maf_total_EUR, breaks = breaks, include.lowest = T, right = F)
gnomad$bin_AFR <- cut(gnomad$maf_total_AFR, breaks = breaks, include.lowest = T, right = F)

gnomad <- as.data.frame(gnomad)
```

```{r}
table(gnomad$bin_EUR)
table(gnomad$bin_AFR)
```
Create the correction data dataframe
```{r}
corr_data_EUR <- data.frame(CHR = gnomad$CHROM,
                            POS = gnomad$POS,
                            proxy_MAF = gnomad$maf_gnomad_EUR,
                            bin = gnomad$bin_EUR)

corr_data_AFR <- data.frame(CHR = gnomad$CHROM,
                            POS = gnomad$POS,
                            proxy_MAF = gnomad$maf_gnomad_AFR,
                            bin = gnomad$bin_AFR)

corr_data_EUR <- corr_data_EUR[corr_data_EUR$proxy_MAF > 0, ]
corr_data_AFR <- corr_data_AFR[corr_data_AFR$proxy_MAF > 0, ]
```

```{r}
gnomad$OR_EUR <- exp(gnomad$beta_EUR)
gnomad$OR_AFR <- exp(gnomad$beta_AFR)
```

```{r}
n_var <- c(100, 500, 1000, 5000, 10000, 15000, 30000, 50000)
```

# keep only needed columns
```{r}
gnomad <- gnomad %>% select(CHROM, POS, REF, ALT, pos37, match_AFnfe, match_AFafr, 
                            maf_cases_EUR, maf_cases_AFR, maf_controls_EUR, maf_controls_AFR,
                            maf_total_EUR, maf_total_AFR, maf_gnomad_EUR, maf_gnomad_AFR, bin_EUR,
                            bin_AFR, OR_EUR, OR_AFR, se_EUR, se_AFR)
```


# EUR test first
```{r}
nCase_diab = 16550
nControl_diab = 403923

gnomad <- as.data.frame(gnomad)

set.seed(2024)

for(i in 1:length(n_var)) {
  for(rep in 1:10) {
    print(paste0(n_var[i], " vars per bin rep: ", rep))
    
    corr_data_sub <- corr_data_EUR %>%
    group_by(bin) %>%
    slice(sample(1:n(), n_var[i]))
    
    #print(summary(corr_data_sub))
    
    res <- CaseControl_SE(data = gnomad[gnomad$maf_total_EUR > 0, ],
                          OR_colname = "OR_EUR",
                          SE_colname = "se_EUR",
                          chromosome_colname = "CHROM",
                          position_colname = "POS",
                          sex_chromosomes = F,
                          N_case = nCase_diab,
                          N_control = nControl_diab,
                          do_correction = T,
                          correction_data = corr_data_sub)
    
    res$bias_unadj <- res$MAF_total - res$maf_total_EUR
    res$bias <- res$MAF_total_adj - res$maf_total_EUR
    
    res_summary <- res %>% group_by(bin_EUR) %>% 
      summarise(mean_bias = mean(bias),
                max_bias = max(abs(bias)),
                median_bias = median(bias))
    res_summary <- as.data.frame(res_summary)
    res_summary$n_var <- n_var[i]
    res_summary$rep <- rep
    
    if(i == 1 & rep == 1) {
      res_summary_unadj <- as.data.frame(res %>% 
                                     group_by(bin_EUR) %>% 
                                     summarise(mean_bias = mean(bias_unadj),
                                     max_bias = max(abs(bias_unadj)),
                                     median_bias = median(bias_unadj)))
      res_summary_unadj$n_var <- 0
      res_summary_unadj$rep <- 0
      res_EUR_all <- res_summary_unadj
      res_EUR_all <- rbind(res_EUR_all, res_summary)
    } else {
      res_EUR_all <- rbind(res_EUR_all, res_summary)
    }
  }
}

saveRDS(res_EUR_all, "CCAFE_correction_testing.rds")
```

```{r}
res_EUR_all
```
```{r}
library(viridis)
```

```{r}
# get the lines dataframe
ref_lines <- res_EUR_all %>%
  filter(n_var == 0, rep == 0) %>%
  select(bin_EUR, mean_bias)

res_EUR_all_plot <- res_EUR_all %>% filter(rep != 0)

# plot with legend first to save legend
p_mean <- ggplot(res_EUR_all_plot, #%>% filter(bin_EUR != "[0,0.1)") , 
       aes(x = bin_EUR, y = mean_bias, fill = as.factor(n_var))) +
  geom_boxplot() +
  scale_fill_manual(values = viridis(8)) +
  geom_segment(data = ref_lines, inherit.aes = F,
               aes(x = as.numeric(as.factor(bin_EUR))  - 0.4,  # Align left
                   xend = as.numeric(as.factor(bin_EUR)) + 0.4,  # Align right 
                   y = mean_bias, yend = mean_bias), 
               color = "red", linetype = "dashed", size = 1) +
  theme_bw() +
  labs(x = "MAF Bin", y = "Mean Bias", fill = "Number of Variants") +
  theme(legend.title = element_text(size = 16),
        legend.text = element_text(size = 14))

legend <- get_legend(p_mean)

# mean bias plot with no legend
p_mean <- ggplot(res_EUR_all_plot %>% filter(bin_EUR != "[0,0.1)") , 
       aes(x = bin_EUR, y = mean_bias, fill = as.factor(n_var))) +
  geom_boxplot() +
  scale_fill_manual(values = viridis(8)) +
  geom_segment(data = ref_lines[2:5, ], inherit.aes = F,
               aes(x = as.numeric(as.factor(bin_EUR)) - 1 - 0.4,  # Align left
                   xend = as.numeric(as.factor(bin_EUR))- 1 + 0.4,  # Align right 
                   y = mean_bias, yend = mean_bias), 
               color = "grey", linetype = "dotted", size = 0.8) +
  theme_bw() +
  coord_cartesian(ylim = c(-0.15, 0.025)) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "red") +
  labs(x = "MAF Bin", y = "Mean Bias", fill = "Variants per bin") +
  theme(legend.position = "none",
        axis.text = element_text(size = 14),
        axis.title = element_text(size = 16))

# get lines dataframe for median bias
ref_lines <- res_EUR_all %>%
  filter(n_var == 0, rep == 0) %>%
  select(bin_EUR, median_bias)

# plot median bias (no legend)
p_median <- ggplot(res_EUR_all_plot %>% filter(bin_EUR != "[0,0.1)") , 
       aes(x = bin_EUR, y = median_bias, fill = as.factor(n_var))) +
  geom_boxplot() +
  scale_fill_manual(values = viridis(8)) +
  geom_segment(data = ref_lines[2:5, ], inherit.aes = F,
               aes(x = as.numeric(as.factor(bin_EUR)) - 1 - 0.4,  # Align left
                   xend = as.numeric(as.factor(bin_EUR))- 1 + 0.4,  # Align right 
                   y = median_bias, yend = median_bias), 
               color = "grey", linetype = "dotted", size = 0.8) +
  theme_bw() +
  coord_cartesian(ylim = c(-0.15, 0.025)) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "red") +
  labs(x = "MAF Bin", y = "Median Bias", fill = "Variants per bin") +
  theme(legend.position = "none",
        axis.text = element_text(size = 14),
        axis.title = element_text(size = 16))

# combine plots
plot <- cowplot::plot_grid(p_mean, p_median, nrow = 2)

# add back in legend and title
final_plot <- cowplot::plot_grid(
  cowplot::ggdraw() + 
    draw_label("EUR Bias Correction - 10 Replicates", 
               fontface = "bold", size = 25, hjust = 0.5), 
  cowplot::plot_grid(plot, legend, ncol = 2, rel_widths = c(1, 0.25)), 
  ncol = 1, 
  rel_heights = c(0.1, 1)  # Adjust title height
)

final_plot

ggsave(filename = "CCAFE_correction_EUR.png", width = 14, height = 10, units = "in", device = "png")
```

# repeat for AFR
```{r}
set.seed(200)

nCase_diab = 668
nControl_diab = 5956

for(i in 1:length(n_var)) {
  for(rep in 1:10) {
    print(paste0(n_var[i], " vars per bin rep: ", rep))
    
    corr_data_sub <- corr_data_AFR %>%
    group_by(bin) %>%
    slice(sample(1:n(), n_var[i]))
    
    #print(summary(corr_data_sub))
    
    res <- CaseControl_SE(data = gnomad,
                          OR_colname = "OR_AFR",
                          SE_colname = "se_AFR",
                          chromosome_colname = "CHROM",
                          position_colname = "POS",
                          sex_chromosomes = F,
                          N_case = nCase_diab,
                          N_control = nControl_diab,
                          do_correction = T,
                          correction_data = corr_data_sub)
    
    res$bias_unadj <- res$MAF_total - res$maf_total_AFR
    res$bias <- res$MAF_total_adj - res$maf_total_AFR
    
    res_summary <- res %>% group_by(bin_AFR) %>% 
      summarise(mean_bias = mean(bias),
                max_bias = max(abs(bias)),
                median_bias = median(bias))
    res_summary <- as.data.frame(res_summary)
    res_summary$n_var <- n_var[i]
    res_summary$rep <- rep
    
    if(i == 1 & rep == 1) {
      res_summary_unadj <- as.data.frame(res %>% 
                                     group_by(bin_AFR) %>% 
                                     summarise(mean_bias = mean(bias_unadj),
                                     max_bias = max(abs(bias_unadj)),
                                     median_bias = median(bias_unadj)))
      res_summary_unadj$n_var <- 0
      res_summary_unadj$rep <- 0
      res_AFR_all <- res_summary_unadj
      res_AFR_all <- rbind(res_AFR_all, res_summary)
    } else {
      res_AFR_all <- rbind(res_AFR_all, res_summary)
    }
  }
}

saveRDS(res_AFR_all, "CCAFE_correction_testing_AFR.rds")
```

```{r}
# get the lines dataframe
ref_lines <- res_AFR_all %>%
  filter(n_var == 0, rep == 0) %>%
  select(bin_AFR, mean_bias)

res_AFR_all_plot <- res_AFR_all %>% filter(rep != 0)

# plot with legend first to save legend
p_mean <- ggplot(res_AFR_all_plot, #%>% filter(bin_AFR != "[0,0.1)") , 
       aes(x = bin_AFR, y = mean_bias, fill = as.factor(n_var))) +
  geom_boxplot() +
  scale_fill_manual(values = viridis(8)) +
  geom_segment(data = ref_lines, inherit.aes = F,
               aes(x = as.numeric(as.factor(bin_AFR))  - 0.4,  # Align left
                   xend = as.numeric(as.factor(bin_AFR)) + 0.4,  # Align right 
                   y = mean_bias, yend = mean_bias), 
               color = "red", linetype = "dashed", size = 1) +
  theme_bw() +
  labs(x = "MAF Bin", y = "Mean Bias", fill = "Number of Variants") +
  theme(legend.title = element_text(size = 16),
        legend.text = element_text(size = 14))

legend <- get_legend(p_mean)

# mean bias plot with no legend
p_mean <- ggplot(res_AFR_all_plot %>% filter(bin_AFR != "[0,0.1)") , 
       aes(x = bin_AFR, y = mean_bias, fill = as.factor(n_var))) +
  geom_boxplot() +
  scale_fill_manual(values = viridis(8)) +
  geom_segment(data = ref_lines[2:5, ], inherit.aes = F,
               aes(x = as.numeric(as.factor(bin_AFR)) - 1 - 0.4,  # Align left
                   xend = as.numeric(as.factor(bin_AFR))- 1 + 0.4,  # Align right 
                   y = mean_bias, yend = mean_bias), 
               color = "grey", linetype = "dotted", size = 0.8) +
  theme_bw() +
  coord_cartesian(ylim = c(-0.15, 0.025)) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "red") +
  labs(x = "MAF Bin", y = "Mean Bias", fill = "Variants per bin") +
  theme(legend.position = "none",
        axis.text = element_text(size = 14),
        axis.title = element_text(size = 16))

# get lines dataframe for median bias
ref_lines <- res_AFR_all %>%
  filter(n_var == 0, rep == 0) %>%
  select(bin_AFR, median_bias)

# plot median bias (no legend)
p_median <- ggplot(res_AFR_all_plot %>% filter(bin_AFR != "[0,0.1)") , 
       aes(x = bin_AFR, y = median_bias, fill = as.factor(n_var))) +
  geom_boxplot() +
  scale_fill_manual(values = viridis(8)) +
  geom_segment(data = ref_lines[2:5, ], inherit.aes = F,
               aes(x = as.numeric(as.factor(bin_AFR)) - 1 - 0.4,  # Align left
                   xend = as.numeric(as.factor(bin_AFR))- 1 + 0.4,  # Align right 
                   y = median_bias, yend = median_bias), 
               color = "grey", linetype = "dotted", size = 0.8) +
  theme_bw() +
  coord_cartesian(ylim = c(-0.15, 0.025)) +
  geom_hline(yintercept = 0, linetype = "dashed", color = "red") +
  labs(x = "MAF Bin", y = "Median Bias", fill = "Variants per bin") +
  theme(legend.position = "none",
        axis.text = element_text(size = 14),
        axis.title = element_text(size = 16))

# combine plots
plot <- cowplot::plot_grid(p_mean, p_median, nrow = 2)

# add back in legend and title
final_plot <- cowplot::plot_grid(
  cowplot::ggdraw() + 
    draw_label("AFR Bias Correction - 10 Replicates", 
               fontface = "bold", size = 25, hjust = 0.5), 
  cowplot::plot_grid(plot, legend, ncol = 2, rel_widths = c(1, 0.25)), 
  ncol = 1, 
  rel_heights = c(0.1, 1)  # Adjust title height
)

final_plot

ggsave(filename = "CCAFE_correction_AFR.png", width = 14, height = 10, units = "in", device = "png")
```

