<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>CCAFE Vignette • CCAFE</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="CCAFE Vignette">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CCAFE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/CCAFE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CCAFE_CCGWAS.html">CCAFE CC-GWAS Example</a>
    </li>
    <li>
      <a href="../articles/CCAFE_Extra_Details.html">CCAFE Extra Details</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/wolffha/CCAFE/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>CCAFE Vignette</h1>
                        <h4 data-toc-skip class="author">Hayley
Stoneman</h4>
            
            <h4 data-toc-skip class="date">2025-03-10</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/wolffha/CCAFE/blob/HEAD/vignettes/CCAFE.Rmd" class="external-link"><code>vignettes/CCAFE.Rmd</code></a></small>
      <div class="hidden name"><code>CCAFE.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><em>Motivation</em></p>
<p>Growth in the field of genetic and genomic research has vastly
increased the amount of available data which is often made publicly
available through summary statistics. While there are many opportunities
and new methods to use summary-level genetic data, it is often limited
by the statistics that are made available. Many post-hoc analyses of
disease data require case and control allele frequencies (AFs), which
are not always published. We present a framework to derive the case and
control AFs from Genome Wide Association Study (GWAS) summary statistics
using the whole sample (case and control aggregated) AF, odds ratio, and
case and control sample sizes, and compare it to a previously published
method that uses the standard error (SE), odds ratio, and case and
control sample sizes to recapitulate case and control AFs.</p>
<p><em>Results</em></p>
<p>In simulations and real data, we find estimating case and control AFs
using the whole sample AF is highly accurate across all settings. When
applying this method to the Pan-UK Biobank we find high concordance with
the known AFs. Conversely, we find that in Pan-UK Biobank and
simulations including covariates, deriving case and control AFs from the
SE underestimates the minor AF (MAF) for higher MAFs. To enable
estimation of case and control AFs using SE, we use gnomAD v3.1.2 AFs as
a proxy for true AFs to estimate and correct for bias fit. While
estimating the case control AF using the whole sample AF is preferred
due to its high accuracy, estimating from the SE can be used more
broadly since the SE can be calculated from the p-value and beta
estimate, which are more commonly provided. Both methods expand the
utility of publicly available genetic summary statistics, can be used
when different statistics are reported, and promote the reusability of
genomic data.</p>
<p><br><br> This document introduces you to the functions in CCAFE and
gives small examples of how they can be used with summary
statistics.</p>
<p><br><br> Find more details and documentation on GitHub: (<a href="https://wolffha.github.io/CCAFE_documentation/" class="external-link uri">https://wolffha.github.io/CCAFE_documentation/</a>)</p>
<p><br><br> To jump forward to individual function specifics:</p>
<p><a href="#CaseControl_AF"><strong>CaseControl_AF</strong></a> – <a href="#a-quick-demo-of-CaseControl_AF">fast forward to example</a></p>
<p><a href="#CaseControl_SE"><strong>CaseControl_SE</strong></a> – <a href="#a-quick-demo-of-CaseControl_SE">fast forward to example</a>
<br><br><br></p>
<p>To view details on using GWAS VCF formatted data and Bioconductor
packages to prepare your data for use in CCAFE:</p>
<p><a href="#VCF">Fast forward to demo</a></p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install this package from BioConductor:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"CCAFE"</span><span class="op">)</span></span></code></pre></div>
<p>To install this package from GitHub:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"https://github.com/wolffha/CCAFE"</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="overview-of-ccafe-functions">Overview of CCAFE Functions<a class="anchor" aria-label="anchor" href="#overview-of-ccafe-functions"></a>
</h3>
<p>CCAFE contains two functions:</p>
<ul>
<li><strong>CaseControl_AF()</strong></li>
<li><strong>CaseControl_SE()</strong></li>
</ul>
<p>The two functions are both used to reconstruct case and control
allele frequencies (AFs) from genetic summary statistics.</p>
<p>The user will select which function to use based on the available
summary statistics.</p>
<p><strong>CaseControl_AF()</strong> has the following required
parameters:</p>
<ul>
<li>Number of cases</li>
<li>Number of controls</li>
<li>Odds Ratio (OR) or beta coefficient</li>
<li>
<strong>AF</strong> for the whole sample (cases and controls
combined)</li>
</ul>
<p><strong>CaseControl_SE()</strong> has the follwoing required
parameters:</p>
<ul>
<li>Number of cases</li>
<li>Number of controls</li>
<li>Odds Ratio (OR) or beta coefficient</li>
<li>
<strong>SE</strong> of the log(OR) for each variant</li>
</ul>
<p><em>Code adapted from ReACt GroupFreq function available here: (<a href="https://github.com/Paschou-Lab/ReAct/blob/main/GrpPRS_src/CountConstruct.c" class="external-link uri">https://github.com/Paschou-Lab/ReAct/blob/main/GrpPRS_src/CountConstruct.c</a>)</em></p>
</div>
</div>
<div class="section level2">
<h2 id="CaseControl_AF">CaseControl_AF<a class="anchor" aria-label="anchor" href="#CaseControl_AF"></a>
</h2>
<div class="section level3">
<h3 id="casecontrol_af-input">CaseControl_AF() input<a class="anchor" aria-label="anchor" href="#casecontrol_af-input"></a>
</h3>
<ul>
<li>
<strong>data</strong>: a dataframe with each row being a variant and
columns for total AF and OR</li>
<li>
<strong>N_case</strong>: an integer for the number of case
samples</li>
<li>
<strong>N_control</strong>: an integer for the number of control
samples</li>
<li>
<strong>OR_colname</strong>: a string containing the exact column
name in <em>data</em> with the odds ratios</li>
<li>
<strong>AF_total_colname</strong>: a string containing the exact
column name in <em>data</em> with the total AFs</li>
</ul>
</div>
<div class="section level3">
<h3 id="casecontrol_af-output">CaseControl_AF() output<a class="anchor" aria-label="anchor" href="#casecontrol_af-output"></a>
</h3>
<p>Returns the original <em>data</em> dataframe with two additional
columns:</p>
<ul>
<li>AF_case</li>
<li>AF_control</li>
</ul>
<p>The number of rows is equal to the number of variants.</p>
</div>
</div>
<div class="section level2">
<h2 id="CaseControl_SE">CaseControl_SE<a class="anchor" aria-label="anchor" href="#CaseControl_SE"></a>
</h2>
<div class="section level3">
<h3 id="casecontrol_se-input">CaseControl_SE() input<a class="anchor" aria-label="anchor" href="#casecontrol_se-input"></a>
</h3>
<p>CaseControl_SE has the following required inputs:</p>
<ul>
<li>
<strong>data</strong>: a dataframe where each row is a variant and
columns for the OR, SE, chromosome, and position</li>
<li>
<strong>N_case</strong>: an integer for the number of case
samples</li>
<li>
<strong>N_control</strong>: an integer for the number of control
samples</li>
<li>
<strong>OR_colname</strong>: a string containing the exact column
name in <em>data</em> with the odds ratios</li>
<li>
<strong>SE_colname</strong>: a string containing the exact column
name in <em>data</em> with the standard errors</li>
<li>
<strong>position_colname</strong>: a string containing the exact
column name in <em>data</em> with the positions of the variants</li>
<li>
<strong>chromosome_colname</strong>: a string containing the exact
column name in <em>data</em> with the chromosome of the variants. Note,
sex chromosomes can be either characters (‘X’, ‘x’, ‘Y’, ‘y’) or numeric
where X=23 and Y=24</li>
<li>
<strong>sex_chromosomes</strong>: boolean, TRUE if variants from sex
chromosome(s) are included in the dataset</li>
<li>
<strong>do_correction</strong>: boolean, TRUE if data is provided to
correct the estimates using proxy MAFs</li>
<li>
<strong>remove_sex_chromosomes</strong>: boolean, TRUE if variants
on sex chromosomes should be removed. This is only necessary if
<em>sex_chromosomes</em> == TRUE and the number of XX/XY individuals per
case and control sample is NOT known</li>
</ul>
<p>CaseControl_SE has the following optional inputs:</p>
<p>If <em>sex_chromosomes</em> == TRUE and
<em>remove_sex_chromosomes</em> == FALSE, the following inputs are
required:</p>
<ul>
<li>
<strong>N_XX_case</strong>: the number of XX chromosome case
individuals</li>
<li>
<strong>N_XX_control</strong>: the number of XX chromosome control
individuals</li>
<li>
<strong>N_XY_case</strong>: the number of XY chromosome case
individuals</li>
<li>
<strong>N_XY_control</strong>: the number of XY chromosome control
individuals</li>
</ul>
<p>If <em>do_correction</em> == TRUE, then data must be provided that
includes harmonized data with proxy MAFs</p>
<ul>
<li>
<strong>correction_data</strong>: a dataframe with the following
EXACT column names: CHR, POS, proxy_MAF, containing data for variants
harmonized between the observed and proxy datasets</li>
</ul>
</div>
<div class="section level3">
<h3 id="casecontrol_se-output">CaseControl_SE() output<a class="anchor" aria-label="anchor" href="#casecontrol_se-output"></a>
</h3>
<p>Returns the <em>data</em> dataframe with three additional
columns:</p>
<ul>
<li>MAF_case</li>
<li>MAF_control</li>
<li>MAF_total</li>
</ul>
<p>Each columns contains the estimated MAF in the cases, controls, and
whole sample, respectively. The number of rows is equal to the number of
variants</p>
<p>If <em>do_correction</em> == TRUE, three additional columns are
included in the dataframe:</p>
<ul>
<li>MAF_case_adj</li>
<li>MAF_control_adj</li>
<li>MAF_total_adj</li>
</ul>
<p>Containing MAFs adjusted using the the proxy MAFs to model the
expected bias.</p>
<p><em>NOTE:</em> This method assumes we are estimating the minor allele
frequency (MAF). The minor allele may or may not be the effect variant
used to calculate AF for the GWAS summary statistics. For additional
details see vignette titled “CCAFE Extra Details”</p>
</div>
</div>
<div class="section level2">
<h2 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h2>
<p>These examples use the provided sample data which is a subset of 500
variants from chromosome 1 of the Pan-UKBB diabetes GWAS in non-Finnish
European individuals.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/wolffha/CCAFE/" class="external-link">CCAFE</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="a-quick-demo-of-CaseControl_AF">A quick demo of CaseControl_AF()<a class="anchor" aria-label="anchor" href="#a-quick-demo-of-CaseControl_AF"></a>
</h3>
<p>Here is a quick demonstration of using the CaseControl_AF function in
the CCAFE package using the sample data. This code block loads the
library. It then loads the sample data and ensures that it is in the
required format (dataframe). It then runs the methods using the required
parameters (data, N_case, N_control, OR_colname, AF_total_colname) and
prints out the first few lines of the resulting returned dataframe.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load the data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"sampleDat"</span><span class="op">)</span></span>
<span><span class="va">sampleDat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">sampleDat</span><span class="op">)</span></span>
<span></span>
<span><span class="va">results_af</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CaseControl_AF.html">CaseControl_AF</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">sampleDat</span>,</span>
<span>                             N_case <span class="op">=</span> <span class="fl">16550</span>,</span>
<span>                             N_control <span class="op">=</span> <span class="fl">403923</span>,</span>
<span>                             OR_colname <span class="op">=</span> <span class="st">"OR"</span>,</span>
<span>                             AF_total_colname <span class="op">=</span> <span class="st">"true_maf_pop"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">results_af</span><span class="op">)</span></span>
<span><span class="co">#&gt;    CHR       POS REF ALT true_maf_case true_maf_control       beta      SE</span></span>
<span><span class="co">#&gt; 1 chr1 226824710   C   T     2.048e-01        2.041e-01  0.0003441 0.01471</span></span>
<span><span class="co">#&gt; 2 chr1 117812346   G   A     4.480e-01        4.471e-01 -0.0013920 0.01196</span></span>
<span><span class="co">#&gt; 3 chr1 230838863   C   T     1.647e-01        1.642e-01  0.0042250 0.01620</span></span>
<span><span class="co">#&gt; 4 chr1  93121792 ATT   A     0.000e+00        0.000e+00 -0.1058000 3.53000</span></span>
<span><span class="co">#&gt; 5 chr1 240236388   G   A     1.114e-05        3.775e-05 -1.2540000 1.25300</span></span>
<span><span class="co">#&gt; 6 chr1  12385196   G   A     1.390e-02        1.402e-02 -0.0152700 0.05046</span></span>
<span><span class="co">#&gt;    gnomad_maf        OR true_maf_pop      AF_case   AF_control</span></span>
<span><span class="co">#&gt; 1 2.04050e-01 1.0003442 2.041276e-01 2.041813e-01 2.041254e-01</span></span>
<span><span class="co">#&gt; 2 4.45508e-01 0.9986090 4.471354e-01 4.468049e-01 4.471490e-01</span></span>
<span><span class="co">#&gt; 3 1.82612e-01 1.0042339 1.642197e-01 1.647775e-01 1.641968e-01</span></span>
<span><span class="co">#&gt; 4 1.07925e-04 0.8996046 0.000000e+00 0.000000e+00 0.000000e+00</span></span>
<span><span class="co">#&gt; 5 8.82145e-05 0.2853611 3.670262e-05 1.077692e-05 3.776488e-05</span></span>
<span><span class="co">#&gt; 6 1.15995e-02 0.9848460 1.401528e-02 1.381395e-02 1.402353e-02</span></span></code></pre></div>
<p>We can then plot the estimates from this example using the provided
known AFs.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot the results</span></span>
<span><span class="co"># first need minor allele frequency to compare to true_maf_case</span></span>
<span><span class="va">results_af</span><span class="op">$</span><span class="va">MAF_case</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">results_af</span><span class="op">$</span><span class="va">AF_case</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;</span> <span class="fl">0.5</span>, <span class="fl">1</span><span class="op">-</span><span class="va">x</span>, <span class="va">x</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">results_af</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">results_af</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">results_af</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">true_maf_case</span>, y <span class="op">=</span> <span class="va">MAF_case</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_abline</a></span><span class="op">(</span>slope <span class="op">=</span> <span class="fl">1</span>, intercept <span class="op">=</span> <span class="fl">0</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span>, color <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Example CaseControl_AF()"</span><span class="op">)</span></span></code></pre></div>
<p><img src="CCAFE_files/figure-html/unnamed-chunk-5-1.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="a-quick-demo-of-CaseControl_SE">A quick demo of CaseControl_SE()<a class="anchor" aria-label="anchor" href="#a-quick-demo-of-CaseControl_SE"></a>
</h3>
<p>Here is a quick demonstration of using the CaseControl_SE function in
the CCAFE package using the sample data. This code block loads the
library. It then loads the sample data and ensures that it is in the
required format (dataframe). It then runs the methods using the required
parameters (data, N_case, N_control, OR_colname, SE_colname,
chromosome_colname, position_colname, do_correction, sex_chromosomes)
and prints out the first few lines of the resulting returned
dataframe.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load the data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"sampleDat"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># always ensure data is a dataframe before inputting to CCAFE methods</span></span>
<span><span class="va">sampleDat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">sampleDat</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># First run without correction</span></span>
<span><span class="va">results_se_noCorr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CaseControl_SE.html">CaseControl_SE</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">sampleDat</span>,</span>
<span>                                    N_case <span class="op">=</span> <span class="fl">16550</span>,</span>
<span>                                    N_control <span class="op">=</span> <span class="fl">403923</span>,</span>
<span>                                    OR_colname <span class="op">=</span> <span class="st">"OR"</span>,</span>
<span>                                    SE_colname <span class="op">=</span> <span class="st">"SE"</span>,</span>
<span>                                    chromosome_colname <span class="op">=</span> <span class="st">"CHR"</span>,</span>
<span>                                    position_colname <span class="op">=</span> <span class="st">"POS"</span>,</span>
<span>                                    do_correction <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                    sex_chromosomes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">results_se_noCorr</span><span class="op">)</span></span>
<span><span class="co">#&gt;    CHR       POS REF ALT true_maf_case true_maf_control       beta      SE</span></span>
<span><span class="co">#&gt; 1 chr1 226824710   C   T     2.048e-01        2.041e-01  0.0003441 0.01471</span></span>
<span><span class="co">#&gt; 2 chr1 117812346   G   A     4.480e-01        4.471e-01 -0.0013920 0.01196</span></span>
<span><span class="co">#&gt; 3 chr1 230838863   C   T     1.647e-01        1.642e-01  0.0042250 0.01620</span></span>
<span><span class="co">#&gt; 4 chr1  93121792 ATT   A     0.000e+00        0.000e+00 -0.1058000 3.53000</span></span>
<span><span class="co">#&gt; 5 chr1 240236388   G   A     1.114e-05        3.775e-05 -1.2540000 1.25300</span></span>
<span><span class="co">#&gt; 6 chr1  12385196   G   A     1.390e-02        1.402e-02 -0.0152700 0.05046</span></span>
<span><span class="co">#&gt;    gnomad_maf        OR true_maf_pop     MAF_case  MAF_control    MAF_total</span></span>
<span><span class="co">#&gt; 1 2.04050e-01 1.0003442 2.041276e-01 1.764905e-01 1.764405e-01 1.764425e-01</span></span>
<span><span class="co">#&gt; 2 4.45508e-01 0.9986090 4.471354e-01 3.263835e-01 3.266896e-01 3.266776e-01</span></span>
<span><span class="co">#&gt; 3 1.82612e-01 1.0042339 1.642197e-01 1.392353e-01 1.387297e-01 1.387496e-01</span></span>
<span><span class="co">#&gt; 4 1.07925e-04 0.8996046 0.000000e+00 2.513875e-06 2.794421e-06 2.783379e-06</span></span>
<span><span class="co">#&gt; 5 8.82145e-05 0.2853611 3.670262e-05 1.946826e-05 6.821993e-05 6.630105e-05</span></span>
<span><span class="co">#&gt; 6 1.15995e-02 0.9848460 1.401528e-02 1.250050e-02 1.269041e-02 1.268294e-02</span></span></code></pre></div>
<p>We can then plot the estimates from this example using the provided
known AFs.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot the results</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">results_se_noCorr</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">true_maf_case</span>, y <span class="op">=</span> <span class="va">MAF_case</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_abline</a></span><span class="op">(</span>slope <span class="op">=</span> <span class="fl">1</span>, intercept <span class="op">=</span> <span class="fl">0</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span>, color <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html" class="external-link">coord_cartesian</a></span><span class="op">(</span>ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">.5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Example CaseControl_SE() No Correction"</span><span class="op">)</span></span></code></pre></div>
<p><img src="CCAFE_files/figure-html/unnamed-chunk-6-1.png" width="100%"></p>
<p>Here is a quick demonstration of using the CaseControl_SE function in
the CCAFE package using the sample data, but also performing the
provided bias correction. This code block loads the library. It then
loads the sample data and ensures that it is in the required format
(dataframe). It then runs the methods using the required parameters
(data, N_case, N_control, OR_colname, SE_colname, chromosome_colname,
position_colname, do_correction, sex_chromosomes) and prints out the
first few lines of the resulting returned dataframe.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load the data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"sampleDat"</span><span class="op">)</span></span>
<span><span class="co"># always ensure data is a dataframe before inputting to CCAFE methods</span></span>
<span><span class="va">sampleDat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">sampleDat</span><span class="op">)</span></span>
<span></span>
<span><span class="va">corr_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>CHR <span class="op">=</span> <span class="va">sampleDat</span><span class="op">$</span><span class="va">CHR</span>,</span>
<span>                        POS <span class="op">=</span> <span class="va">sampleDat</span><span class="op">$</span><span class="va">POS</span>,</span>
<span>                        proxy_MAF <span class="op">=</span> <span class="va">sampleDat</span><span class="op">$</span><span class="va">gnomad_maf</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># now run with correction</span></span>
<span><span class="va">results_se_corr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CaseControl_SE.html">CaseControl_SE</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">sampleDat</span>,</span>
<span>                                  N_case <span class="op">=</span> <span class="fl">16550</span>,</span>
<span>                                  N_control <span class="op">=</span> <span class="fl">403923</span>,</span>
<span>                                  OR_colname <span class="op">=</span> <span class="st">"OR"</span>,</span>
<span>                                  SE_colname <span class="op">=</span> <span class="st">"SE"</span>,</span>
<span>                                  chromosome_colname <span class="op">=</span> <span class="st">"CHR"</span>,</span>
<span>                                  position_colname <span class="op">=</span> <span class="st">"POS"</span>,</span>
<span>                                  do_correction <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                  correction_data <span class="op">=</span> <span class="va">corr_data</span>,</span>
<span>                                  sex_chromosomes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">results_se_corr</span><span class="op">)</span></span>
<span><span class="co">#&gt;    CHR       POS REF ALT true_maf_case true_maf_control       beta      SE</span></span>
<span><span class="co">#&gt; 1 chr1 226824710   C   T     2.048e-01        2.041e-01  0.0003441 0.01471</span></span>
<span><span class="co">#&gt; 2 chr1 117812346   G   A     4.480e-01        4.471e-01 -0.0013920 0.01196</span></span>
<span><span class="co">#&gt; 3 chr1 230838863   C   T     1.647e-01        1.642e-01  0.0042250 0.01620</span></span>
<span><span class="co">#&gt; 4 chr1  93121792 ATT   A     0.000e+00        0.000e+00 -0.1058000 3.53000</span></span>
<span><span class="co">#&gt; 5 chr1 240236388   G   A     1.114e-05        3.775e-05 -1.2540000 1.25300</span></span>
<span><span class="co">#&gt; 6 chr1  12385196   G   A     1.390e-02        1.402e-02 -0.0152700 0.05046</span></span>
<span><span class="co">#&gt;    gnomad_maf        OR true_maf_pop     MAF_case  MAF_control    MAF_total</span></span>
<span><span class="co">#&gt; 1 2.04050e-01 1.0003442 2.041276e-01 1.764905e-01 1.764405e-01 1.764425e-01</span></span>
<span><span class="co">#&gt; 2 4.45508e-01 0.9986090 4.471354e-01 3.263835e-01 3.266896e-01 3.266776e-01</span></span>
<span><span class="co">#&gt; 3 1.82612e-01 1.0042339 1.642197e-01 1.392353e-01 1.387297e-01 1.387496e-01</span></span>
<span><span class="co">#&gt; 4 1.07925e-04 0.8996046 0.000000e+00 2.513875e-06 2.794421e-06 2.783379e-06</span></span>
<span><span class="co">#&gt; 5 8.82145e-05 0.2853611 3.670262e-05 1.946826e-05 6.821993e-05 6.630105e-05</span></span>
<span><span class="co">#&gt; 6 1.15995e-02 0.9848460 1.401528e-02 1.250050e-02 1.269041e-02 1.268294e-02</span></span>
<span><span class="co">#&gt;   MAF_total_adj MAF_control_adj MAF_case_adj</span></span>
<span><span class="co">#&gt; 1  2.100692e-01    2.100673e-01 2.101173e-01</span></span>
<span><span class="co">#&gt; 2  4.834218e-01    4.834338e-01 4.831277e-01</span></span>
<span><span class="co">#&gt; 3  1.594029e-01    1.593830e-01 1.598885e-01</span></span>
<span><span class="co">#&gt; 4  2.783379e-06    2.794421e-06 2.513875e-06</span></span>
<span><span class="co">#&gt; 5  6.630105e-05    6.821993e-05 1.946826e-05</span></span>
<span><span class="co">#&gt; 6  9.160387e-03    9.167862e-03 8.977955e-03</span></span></code></pre></div>
<p>Again, we can plot the estimates (corrected) and compare them to the
provided known AFs and see reduced bias.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot the results</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">results_se_corr</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">true_maf_case</span>, y <span class="op">=</span> <span class="va">MAF_case_adj</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_abline</a></span><span class="op">(</span>slope <span class="op">=</span> <span class="fl">1</span>, intercept <span class="op">=</span> <span class="fl">0</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span>, color <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Example CaseControl_SE() With Correction"</span><span class="op">)</span></span></code></pre></div>
<p><img src="CCAFE_files/figure-html/unnamed-chunk-7-1.png" width="100%"></p>
</div>
</div>
<div class="section level2">
<h2 id="integration-with-bioconductor">Integration with Bioconductor<a class="anchor" aria-label="anchor" href="#integration-with-bioconductor"></a>
</h2>
<div class="section level3">
<h3 id="tabular-data">Tabular data<a class="anchor" aria-label="anchor" href="#tabular-data"></a>
</h3>
<p>This software package aims to increase the utility of existing,
publicly available genomic summary data from genome-wide association
studies (GWAS). Summary statistics are still widely underutilized,
despite their wide availability. As such, Bioconductor packages with
methods specifically designed to analyze these statistics are still
being developed.</p>
<p>The most commonly used repository of these summary statistics is the
NHGRI GWAS Catalog (<a href="https://www.ebi.ac.uk/gwas/" class="external-link uri">https://www.ebi.ac.uk/gwas/</a>). From the GWAS catalog, a
user can download full summary statistics through the FTP sites. The
majority of these summary statistics are zipped plain text (usually
space or tab delimited) files. The GWAS Catalog itself has recommended a
tabular, plain-text format as its recommended GWAS summary statistic
format. As such, this tabular data is most simply read into R as a
dataframe structure.</p>
</div>
<div class="section level3">
<h3 id="VCF">VCF data<a class="anchor" aria-label="anchor" href="#VCF"></a>
</h3>
<p>While the many users will download the GWAS files and load them into
R, there are other ways to obtain GWAS summary statistics, including
through the IEU Open GWAS Project (<a href="https://gwas.mrcieu.ac.uk/" class="external-link uri">https://gwas.mrcieu.ac.uk/</a>). The IEU Open GWAS Project
instead uses the GWAS VCF format (<a href="https://doi.org/10.1101/2020.05.29.115824" class="external-link uri">https://doi.org/10.1101/2020.05.29.115824</a>) to store GWAS
summary statistics and ensure alignment to the hg19 reference genome. We
can use an assortment of Bioconductor packages and data structures to
work with GWAS VCF data.</p>
<p>First we will make use of the Bioconductor package
<em>VariantAnnotation</em> to handle VCF data</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"VariantAnnotation"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">VariantAnnotation</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>We will use the VCF from this GWAS of Type 2 Diabetes <a href="https://doi.org/10.1038/s41588-018-0084-1" class="external-link uri">https://doi.org/10.1038/s41588-018-0084-1</a>. A subset of
10,000 variants has been stored within the package and can be
loaded.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"vcf_sample"</span><span class="op">)</span></span></code></pre></div>
<p>We can see the name of the GWAS dataset (refers to the IEU GWAS
database ID name)</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/class.MIAME.html" class="external-link">samples</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCFHeader-class.html" class="external-link">header</a></span><span class="op">(</span><span class="va">vcf_sample</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "ebi-a-GCST007515"</span></span></code></pre></div>
<p>To understand what the data format, we can look view the VCF
object</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vcf_sample</span></span>
<span><span class="co">#&gt; class: CollapsedVCF </span></span>
<span><span class="co">#&gt; dim: 10000 1 </span></span>
<span><span class="co">#&gt; rowRanges(vcf):</span></span>
<span><span class="co">#&gt;   GRanges with 5 metadata columns: paramRangeID, REF, ALT, QUAL, FILTER</span></span>
<span><span class="co">#&gt; info(vcf):</span></span>
<span><span class="co">#&gt;   DataFrame with 1 column: AF</span></span>
<span><span class="co">#&gt; info(header(vcf)):</span></span>
<span><span class="co">#&gt;       Number Type  Description     </span></span>
<span><span class="co">#&gt;    AF A      Float Allele Frequency</span></span>
<span><span class="co">#&gt; geno(vcf):</span></span>
<span><span class="co">#&gt;   List of length 9: ES, SE, LP, AF, SS, EZ, SI, NC, ID</span></span>
<span><span class="co">#&gt; geno(header(vcf)):</span></span>
<span><span class="co">#&gt;       Number Type    Description                                               </span></span>
<span><span class="co">#&gt;    ES A      Float   Effect size estimate relative to the alternative allele   </span></span>
<span><span class="co">#&gt;    SE A      Float   Standard error of effect size estimate                    </span></span>
<span><span class="co">#&gt;    LP A      Float   -log10 p-value for effect estimate                        </span></span>
<span><span class="co">#&gt;    AF A      Float   Alternate allele frequency in the association study       </span></span>
<span><span class="co">#&gt;    SS A      Integer Sample size used to estimate genetic effect               </span></span>
<span><span class="co">#&gt;    EZ A      Float   Z-score provided if it was used to derive the EFFECT an...</span></span>
<span><span class="co">#&gt;    SI A      Float   Accuracy score of summary data imputation                 </span></span>
<span><span class="co">#&gt;    NC A      Integer Number of cases used to estimate genetic effect           </span></span>
<span><span class="co">#&gt;    ID 1      String  Study variant identifier</span></span></code></pre></div>
<p>The columns that we want for use in CCAFE are:</p>
<p>From the rowRanges object:</p>
<ul>
<li>seqnames (chromosome)</li>
<li>ranges (position)</li>
<li>REF</li>
<li>ALT</li>
<li>RSID</li>
</ul>
<p>From the geno object:</p>
<ul>
<li>ES (effect size of ALT)</li>
<li>SE</li>
<li>AF (allele frequency of ALT)</li>
</ul>
<p>We then want to convert this VCF formatted dataset into a simple
dataframe to use with CCAFE.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># first we will get the info from GRanges object (position, RSID)</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu">ranges</span><span class="op">(</span><span class="va">vcf_sample</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="va">meta</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">4</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Position"</span>, <span class="st">"RSID"</span><span class="op">)</span></span>
<span><span class="co"># get the chromosome (as a)</span></span>
<span><span class="va">meta</span><span class="op">$</span><span class="va">Chromosome</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">vcf_sample</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># now we can also get the meta data (REF, ALT) from the GRanges object</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">meta</span>, <span class="fu">mcols</span><span class="op">(</span><span class="va">vcf_sample</span><span class="op">)</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># now we will get the info from the geno object</span></span>
<span><span class="va">geno_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  beta <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCF-class.html" class="external-link">geno</a></span><span class="op">(</span><span class="va">vcf_sample</span><span class="op">)</span><span class="op">$</span><span class="va">ES</span><span class="op">)</span>,</span>
<span>  SE <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCF-class.html" class="external-link">geno</a></span><span class="op">(</span><span class="va">vcf_sample</span><span class="op">)</span><span class="op">$</span><span class="va">SE</span><span class="op">)</span>,</span>
<span>  AF <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCF-class.html" class="external-link">geno</a></span><span class="op">(</span><span class="va">vcf_sample</span><span class="op">)</span><span class="op">$</span><span class="va">AF</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">df_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">meta</span>, <span class="va">geno_dat</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">df_data</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Position        RSID Chromosome REF ALT    beta     SE     AF</span></span>
<span><span class="co">#&gt; 1   762320  rs75333668          1   C   T -0.0352 0.0574 0.0078</span></span>
<span><span class="co">#&gt; 2   861349 rs200686669          1   C   T -0.0299 0.1821 0.0034</span></span>
<span><span class="co">#&gt; 3   865545 rs201186828          1   G   A  0.1188 0.1535 0.0041</span></span>
<span><span class="co">#&gt; 4   865584 rs148711625          1   G   A -0.2004 0.0840 0.0219</span></span>
<span><span class="co">#&gt; 5   865625 rs146327803          1   G   A -0.0778 0.2306 0.0046</span></span>
<span><span class="co">#&gt; 6   865628  rs41285790          1   G   A -0.0037 0.0510 0.0049</span></span></code></pre></div>
<p>For CCAFE we will use OR, so let’s create this column by
exponentiating the effect estimate</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df_data</span><span class="op">$</span><span class="va">OR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">exp</a></span><span class="op">(</span><span class="va">df_data</span><span class="op">$</span><span class="va">beta</span><span class="op">)</span></span></code></pre></div>
<p>Let’s now apply the CCAFE methods to get case and control specific
AFs</p>
<p>First, since we have total AF, we’ll use CaseControl_AF</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CaseControl_AF.html">CaseControl_AF</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">df_data</span>,</span>
<span>                              N_case <span class="op">=</span> <span class="fl">48286</span>,</span>
<span>                              N_control <span class="op">=</span> <span class="fl">250671</span>,</span>
<span>                              OR_colname <span class="op">=</span> <span class="st">"OR"</span>,</span>
<span>                              AF_total_colname <span class="op">=</span> <span class="st">"AF"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">df_data</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Position        RSID Chromosome REF ALT    beta     SE     AF        OR</span></span>
<span><span class="co">#&gt; 1   762320  rs75333668          1   C   T -0.0352 0.0574 0.0078 0.9654123</span></span>
<span><span class="co">#&gt; 2   861349 rs200686669          1   C   T -0.0299 0.1821 0.0034 0.9705426</span></span>
<span><span class="co">#&gt; 3   865545 rs201186828          1   G   A  0.1188 0.1535 0.0041 1.1261447</span></span>
<span><span class="co">#&gt; 4   865584 rs148711625          1   G   A -0.2004 0.0840 0.0219 0.8184033</span></span>
<span><span class="co">#&gt; 5   865625 rs146327803          1   G   A -0.0778 0.2306 0.0046 0.9251494</span></span>
<span><span class="co">#&gt; 6   865628  rs41285790          1   G   A -0.0037 0.0510 0.0049 0.9963068</span></span>
<span><span class="co">#&gt;       AF_case  AF_control</span></span>
<span><span class="co">#&gt; 1 0.007574252 0.007843485</span></span>
<span><span class="co">#&gt; 2 0.003315901 0.003416200</span></span>
<span><span class="co">#&gt; 3 0.004523116 0.004018497</span></span>
<span><span class="co">#&gt; 4 0.018530155 0.022549123</span></span>
<span><span class="co">#&gt; 5 0.004309041 0.004656047</span></span>
<span><span class="co">#&gt; 6 0.004884892 0.004902910</span></span></code></pre></div>
<p>We can look at a comparison of the AFs</p>
<p>Create a dataframe to plot</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plotdata_AF</span> <span class="op">&lt;-</span> <span class="va">df_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">AF</span>, <span class="va">AF_case</span>, <span class="va">AF_control</span><span class="op">)</span></span>
<span><span class="va">plotdata_AF</span><span class="op">$</span><span class="va">AF_total</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">plotdata_AF</span><span class="op">$</span><span class="va">AF_case</span><span class="op">*</span><span class="fl">48286</span> <span class="op">+</span> <span class="va">plotdata_AF</span><span class="op">$</span><span class="va">AF_control</span><span class="op">*</span><span class="fl">250671</span><span class="op">)</span><span class="op">/</span><span class="op">(</span><span class="fl">298857</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">plotdata_AF</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"AF(known)"</span></span>
<span><span class="va">plotdata_AF_long</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span><span class="va">plotdata_AF</span>, cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">plotdata_AF</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Plot the results</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">plotdata_AF_long</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">value</span>, fill <span class="op">=</span> <span class="va">name</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.5</span>, outliers <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">name</span>, nrow <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="CCAFE_files/figure-html/unnamed-chunk-17-1.png" width="100%"></p>
<p>These case and control AFs can now be used in further downstream
analyses. For example, if the data was harmonized with a reference
panel, then Summix2 (<a href="https://www.bioconductor.org/packages/release/bioc/html/Summix.html" class="external-link uri">https://www.bioconductor.org/packages/release/bioc/html/Summix.html</a>)
could be used to estimate substructure in the GWAS sample (total sample,
cases, or controls). The AFs could also be used in case-case GWAS (see
<a href="https://wolffha.github.io/CCAFE_documentation/articles/CCAFE_CCGWAS.html" class="external-link uri">https://wolffha.github.io/CCAFE_documentation/articles/CCAFE_CCGWAS.html</a>)
or certain meta-analysis software (e.g. <a href="https://doi.org/10.1038/s41598-022-12185-6" class="external-link uri">https://doi.org/10.1038/s41598-022-12185-6</a>).</p>
<div class="section level4">
<h4 id="vcf-conversion-function">VCF Conversion Function<a class="anchor" aria-label="anchor" href="#vcf-conversion-function"></a>
</h4>
<p>For ease of use of CCAFE with VCF files, we have included a function
‘CCAFE_convertVCF()’.</p>
<p>It performs the steps outlined above, assuming the user has loaded
their VCF into R.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df_data_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CCAFE_convertVCF.html">CCAFE_convertVCF</a></span><span class="op">(</span><span class="va">vcf_sample</span><span class="op">)</span></span>
<span><span class="co">#&gt; Valid VCF object, converting to dataframe for CCAFE...</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df_data_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CaseControl_AF.html">CaseControl_AF</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">df_data_2</span>,</span>
<span>                              N_case <span class="op">=</span> <span class="fl">48286</span>,</span>
<span>                              N_control <span class="op">=</span> <span class="fl">250671</span>,</span>
<span>                              OR_colname <span class="op">=</span> <span class="st">"OR"</span>,</span>
<span>                              AF_total_colname <span class="op">=</span> <span class="st">"AF"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">df_data_2</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Position        RSID Chromosome REF ALT    beta     SE     AF        OR</span></span>
<span><span class="co">#&gt; 1   762320  rs75333668          1   C   T -0.0352 0.0574 0.0078 0.9654123</span></span>
<span><span class="co">#&gt; 2   861349 rs200686669          1   C   T -0.0299 0.1821 0.0034 0.9705426</span></span>
<span><span class="co">#&gt; 3   865545 rs201186828          1   G   A  0.1188 0.1535 0.0041 1.1261447</span></span>
<span><span class="co">#&gt; 4   865584 rs148711625          1   G   A -0.2004 0.0840 0.0219 0.8184033</span></span>
<span><span class="co">#&gt; 5   865625 rs146327803          1   G   A -0.0778 0.2306 0.0046 0.9251494</span></span>
<span><span class="co">#&gt; 6   865628  rs41285790          1   G   A -0.0037 0.0510 0.0049 0.9963068</span></span>
<span><span class="co">#&gt;       AF_case  AF_control</span></span>
<span><span class="co">#&gt; 1 0.007574252 0.007843485</span></span>
<span><span class="co">#&gt; 2 0.003315901 0.003416200</span></span>
<span><span class="co">#&gt; 3 0.004523116 0.004018497</span></span>
<span><span class="co">#&gt; 4 0.018530155 0.022549123</span></span>
<span><span class="co">#&gt; 5 0.004309041 0.004656047</span></span>
<span><span class="co">#&gt; 6 0.004884892 0.004902910</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64</span></span>
<span><span class="co">#&gt; Running under: Windows 11 x64 (build 26100)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=English_United States.utf8 </span></span>
<span><span class="co">#&gt; [2] LC_CTYPE=English_United States.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=English_United States.utf8</span></span>
<span><span class="co">#&gt; [4] LC_NUMERIC=C                          </span></span>
<span><span class="co">#&gt; [5] LC_TIME=English_United States.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/Denver</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] VariantAnnotation_1.52.0    Rsamtools_2.22.0           </span></span>
<span><span class="co">#&gt;  [3] Biostrings_2.74.0           XVector_0.46.0             </span></span>
<span><span class="co">#&gt;  [5] SummarizedExperiment_1.36.0 Biobase_2.66.0             </span></span>
<span><span class="co">#&gt;  [7] GenomicRanges_1.58.0        GenomeInfoDb_1.42.1        </span></span>
<span><span class="co">#&gt;  [9] IRanges_2.40.0              S4Vectors_0.44.0           </span></span>
<span><span class="co">#&gt; [11] MatrixGenerics_1.18.0       matrixStats_1.4.1          </span></span>
<span><span class="co">#&gt; [13] BiocGenerics_0.52.0         lubridate_1.9.4            </span></span>
<span><span class="co">#&gt; [15] forcats_1.0.0               stringr_1.5.1              </span></span>
<span><span class="co">#&gt; [17] dplyr_1.1.4                 purrr_1.0.2                </span></span>
<span><span class="co">#&gt; [19] readr_2.1.5                 tidyr_1.3.1                </span></span>
<span><span class="co">#&gt; [21] tibble_3.2.1                ggplot2_3.5.1              </span></span>
<span><span class="co">#&gt; [23] tidyverse_2.0.0             CCAFE_0.99.3               </span></span>
<span><span class="co">#&gt; [25] BiocStyle_2.34.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] DBI_1.2.3                bitops_1.0-9             rlang_1.1.4             </span></span>
<span><span class="co">#&gt;  [4] magrittr_2.0.3           compiler_4.4.1           RSQLite_2.3.9           </span></span>
<span><span class="co">#&gt;  [7] GenomicFeatures_1.58.0   png_0.1-8                systemfonts_1.1.0       </span></span>
<span><span class="co">#&gt; [10] vctrs_0.6.5              pkgconfig_2.0.3          crayon_1.5.3            </span></span>
<span><span class="co">#&gt; [13] fastmap_1.2.0            labeling_0.4.3           utf8_1.2.4              </span></span>
<span><span class="co">#&gt; [16] rmarkdown_2.29           tzdb_0.4.0               UCSC.utils_1.2.0        </span></span>
<span><span class="co">#&gt; [19] ragg_1.3.3               bit_4.5.0.1              xfun_0.49               </span></span>
<span><span class="co">#&gt; [22] zlibbioc_1.52.0          cachem_1.1.0             jsonlite_1.8.9          </span></span>
<span><span class="co">#&gt; [25] blob_1.2.4               DelayedArray_0.32.0      BiocParallel_1.40.0     </span></span>
<span><span class="co">#&gt; [28] parallel_4.4.1           R6_2.5.1                 bslib_0.8.0             </span></span>
<span><span class="co">#&gt; [31] stringi_1.8.4            rtracklayer_1.66.0       jquerylib_0.1.4         </span></span>
<span><span class="co">#&gt; [34] bookdown_0.42            knitr_1.49               Matrix_1.7-1            </span></span>
<span><span class="co">#&gt; [37] timechange_0.3.0         tidyselect_1.2.1         rstudioapi_0.17.1       </span></span>
<span><span class="co">#&gt; [40] abind_1.4-8              yaml_2.3.10              codetools_0.2-20        </span></span>
<span><span class="co">#&gt; [43] curl_6.0.1               lattice_0.22-6           withr_3.0.2             </span></span>
<span><span class="co">#&gt; [46] KEGGREST_1.46.0          evaluate_1.0.1           desc_1.4.3              </span></span>
<span><span class="co">#&gt; [49] pillar_1.9.0             BiocManager_1.30.25      generics_0.1.3          </span></span>
<span><span class="co">#&gt; [52] RCurl_1.98-1.16          hms_1.1.3                munsell_0.5.1           </span></span>
<span><span class="co">#&gt; [55] scales_1.3.0             glue_1.8.0               tools_4.4.1             </span></span>
<span><span class="co">#&gt; [58] BiocIO_1.16.0            BSgenome_1.74.0          GenomicAlignments_1.42.0</span></span>
<span><span class="co">#&gt; [61] fs_1.6.5                 XML_3.99-0.17            grid_4.4.1              </span></span>
<span><span class="co">#&gt; [64] AnnotationDbi_1.68.0     colorspace_2.1-1         GenomeInfoDbData_1.2.13 </span></span>
<span><span class="co">#&gt; [67] restfulr_0.0.15          cli_3.6.3                textshaping_0.4.1       </span></span>
<span><span class="co">#&gt; [70] fansi_1.0.6              S4Arrays_1.6.0           gtable_0.3.6            </span></span>
<span><span class="co">#&gt; [73] sass_0.4.9               digest_0.6.36            SparseArray_1.6.0       </span></span>
<span><span class="co">#&gt; [76] rjson_0.2.23             htmlwidgets_1.6.4        farver_2.1.2            </span></span>
<span><span class="co">#&gt; [79] memoise_2.0.1            htmltools_0.5.8.1        pkgdown_2.1.1           </span></span>
<span><span class="co">#&gt; [82] lifecycle_1.0.4          httr_1.4.7               bit64_4.5.2</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Hayley Wolff.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
