<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>CaseControl_SE — CaseControl_SE • CCAFE</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="CaseControl_SE — CaseControl_SE"><meta property="og:description" content="This is a function to derive the case, control, and total MAFs from GWAS summary statistics when
the user has access to the sample sizes, and the OR (or beta), and SE for the log(OR) for each variant.
If user has total AF instead of SE use [CCAFE::CaseControl_AF()]
This code uses the GroupFreq function adapted from C from &amp;lt;https://github.com/Paschou-Lab/ReAct/blob/main/GrpPRS_src/CountConstruct.c&amp;gt;"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CCAFE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/CCAFE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/CCAFE_CCGWAS.html">CCAFE CC-GWAS Example</a>
    </li>
    <li>
      <a href="../articles/CCAFE_Extra_Details.html">CCAFE Extra Details</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/wolffha/CCAFE/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>CaseControl_SE</h1>
    <small class="dont-index">Source: <a href="https://github.com/wolffha/CCAFE/blob/HEAD/R/CaseControl_SE.R" class="external-link"><code>R/CaseControl_SE.R</code></a></small>
    <div class="hidden name"><code>CaseControl_SE.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This is a function to derive the case, control, and total MAFs from GWAS summary statistics when
the user has access to the sample sizes, and the OR (or beta), and SE for the log(OR) for each variant.
If user has total AF instead of SE use [CCAFE::CaseControl_AF()]
This code uses the GroupFreq function adapted from C from &lt;https://github.com/Paschou-Lab/ReAct/blob/main/GrpPRS_src/CountConstruct.c&gt;</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">CaseControl_SE</span><span class="op">(</span></span>
<span>  <span class="va">data</span>,</span>
<span>  N_case <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  N_control <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  OR_colname <span class="op">=</span> <span class="st">"OR"</span>,</span>
<span>  SE_colname <span class="op">=</span> <span class="st">"SE"</span>,</span>
<span>  chromosome_colname <span class="op">=</span> <span class="st">"chr"</span>,</span>
<span>  sex_chromosomes <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  position_colname <span class="op">=</span> <span class="st">"pos"</span>,</span>
<span>  N_XX_case <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  N_XX_control <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  N_XY_case <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  N_XY_control <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  do_correction <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  correction_data <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  remove_sex_chromosomes <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-data">data<a class="anchor" aria-label="anchor" href="#arg-data"></a></dt>
<dd><p>dataframe where each row is a variant and columns contain the OR, SE, chromosome and positions</p></dd>


<dt id="arg-n-case">N_case<a class="anchor" aria-label="anchor" href="#arg-n-case"></a></dt>
<dd><p>an integer of the number of Case individuals</p></dd>


<dt id="arg-n-control">N_control<a class="anchor" aria-label="anchor" href="#arg-n-control"></a></dt>
<dd><p>an integer of the number of Control individuals</p></dd>


<dt id="arg-or-colname">OR_colname<a class="anchor" aria-label="anchor" href="#arg-or-colname"></a></dt>
<dd><p>a string containing the exact column name in 'data' with the OR</p></dd>


<dt id="arg-se-colname">SE_colname<a class="anchor" aria-label="anchor" href="#arg-se-colname"></a></dt>
<dd><p>a string containing the exact column name in 'data' with the SE</p></dd>


<dt id="arg-chromosome-colname">chromosome_colname<a class="anchor" aria-label="anchor" href="#arg-chromosome-colname"></a></dt>
<dd><p>a string containing the exact column name in 'data' with the chromosomes, default "chr"</p></dd>


<dt id="arg-sex-chromosomes">sex_chromosomes<a class="anchor" aria-label="anchor" href="#arg-sex-chromosomes"></a></dt>
<dd><p>boolean, TRUE if variants from sex chromosomes are included in the dataset. Sex chromosomes can be numeric (23, 24) or character (X, Y). If numeric, assumes X=23 and Y=24.</p></dd>


<dt id="arg-position-colname">position_colname<a class="anchor" aria-label="anchor" href="#arg-position-colname"></a></dt>
<dd><p>a string containing the exact column name in 'data' with the position, default "pos"</p></dd>


<dt id="arg-n-xx-case">N_XX_case<a class="anchor" aria-label="anchor" href="#arg-n-xx-case"></a></dt>
<dd><p>the number of XX chromosome case individuals (REQUIRED if sex_chromosomes == TRUE)</p></dd>


<dt id="arg-n-xx-control">N_XX_control<a class="anchor" aria-label="anchor" href="#arg-n-xx-control"></a></dt>
<dd><p>the number of XX chromosome control individuals (REQUIRED if sex_chromosomes == TRUE)</p></dd>


<dt id="arg-n-xy-case">N_XY_case<a class="anchor" aria-label="anchor" href="#arg-n-xy-case"></a></dt>
<dd><p>the number of XY chromosome case individuals (REQUIRED if sex_chromosomes == TRUE)</p></dd>


<dt id="arg-n-xy-control">N_XY_control<a class="anchor" aria-label="anchor" href="#arg-n-xy-control"></a></dt>
<dd><p>the number of XY chromosome control individuals (REQUIRED if sex_chromosomes == TRUE)</p></dd>


<dt id="arg-do-correction">do_correction<a class="anchor" aria-label="anchor" href="#arg-do-correction"></a></dt>
<dd><p>boolean, TRUE if data is provided to perform correction</p></dd>


<dt id="arg-correction-data">correction_data<a class="anchor" aria-label="anchor" href="#arg-correction-data"></a></dt>
<dd><p>a dataframe with the following exact columns: CHR, POS, proxy_MAF with data that is harmonized between the proxy true datasets and the observed dataset</p></dd>


<dt id="arg-remove-sex-chromosomes">remove_sex_chromosomes<a class="anchor" aria-label="anchor" href="#arg-remove-sex-chromosomes"></a></dt>
<dd><p>boolean, TRUE if should keep autosomes only. This is needed when the number of biological sex males/females per case and control group is not known.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>returns data as a dataframe with three additional columns: MAF_case, MAF_control, MAF_total for the estimated MAFs for each variant. If do_correction = TRUE, then will output 3 additional columns (MAF_case_adj, MAF_control_adj, MAF_total_adj) with the adjusted estimates.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>https://github.com/wolffha/CCAFE</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><a href="https://github.com/wolffha/CCAFE" class="external-link">https://github.com/wolffha/CCAFE</a> for further documentation</p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Hayley Wolff (Stoneman), <a href="mailto:hayley.wolff@cuanschutz.edu">hayley.wolff@cuanschutz.edu</a></p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/wolffha/CCAFE/" class="external-link">CCAFE</a></span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"sampleDat"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">sampleDat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">sampleDat</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">nCase_sample</span> <span class="op">=</span> <span class="fl">16550</span></span></span>
<span class="r-in"><span><span class="va">nControl_sample</span> <span class="op">=</span> <span class="fl">403923</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># get the estimated case and control MAFs</span></span></span>
<span class="r-in"><span><span class="va">se_method_results</span> <span class="op">&lt;-</span> <span class="fu">CaseControl_SE</span><span class="op">(</span>data <span class="op">=</span> <span class="va">sampleDat</span>,</span></span>
<span class="r-in"><span>                                    N_case <span class="op">=</span> <span class="va">nCase_sample</span>,</span></span>
<span class="r-in"><span>                                    N_control <span class="op">=</span> <span class="va">nControl_sample</span>,</span></span>
<span class="r-in"><span>                                    OR_colname <span class="op">=</span> <span class="st">"OR"</span>,</span></span>
<span class="r-in"><span>                                    SE_colname <span class="op">=</span> <span class="st">"SE"</span>,</span></span>
<span class="r-in"><span>                                    chromosome_colname <span class="op">=</span> <span class="st">"CHR"</span>,</span></span>
<span class="r-in"><span>                                    position_colname <span class="op">=</span> <span class="st">"POS"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">se_method_results</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    CHR       POS REF ALT true_maf_case true_maf_control       beta      SE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 chr1 226824710   C   T     2.048e-01        2.041e-01  0.0003441 0.01471</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 chr1 117812346   G   A     4.480e-01        4.471e-01 -0.0013920 0.01196</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 chr1 230838863   C   T     1.647e-01        1.642e-01  0.0042250 0.01620</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 chr1  93121792 ATT   A     0.000e+00        0.000e+00 -0.1058000 3.53000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 chr1 240236388   G   A     1.114e-05        3.775e-05 -1.2540000 1.25300</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 chr1  12385196   G   A     1.390e-02        1.402e-02 -0.0152700 0.05046</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    gnomad_maf        OR true_maf_pop     MAF_case  MAF_control    MAF_total</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 2.04050e-01 1.0003442 2.041276e-01 1.764905e-01 1.764405e-01 1.764425e-01</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 4.45508e-01 0.9986090 4.471354e-01 3.263835e-01 3.266896e-01 3.266776e-01</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 1.82612e-01 1.0042339 1.642197e-01 1.392353e-01 1.387297e-01 1.387496e-01</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 1.07925e-04 0.8996046 0.000000e+00 2.513875e-06 2.794421e-06 2.783379e-06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 8.82145e-05 0.2853611 3.670262e-05 1.946826e-05 6.821993e-05 6.630105e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 1.15995e-02 0.9848460 1.401528e-02 1.250050e-02 1.269041e-02 1.268294e-02</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hayley Wolff.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer></div>






  </body></html>

